StateNet.Tests/Network/StateNetworkBuilder_Tests.cs

```diff
@@ -1,25 +1,88 @@
-﻿using Aptacode.StateNet.Network;
+﻿using Aptacode.Expressions;
+using Aptacode.Expressions.Integer;
+using Aptacode.StateNet.Engine.Transitions;
+using Aptacode.StateNet.Network;
+using Moq;
+using StateNet.Tests.Network.Helpers;
+using System.Collections.Generic;
+using System.Linq;
 using Xunit;
 
 namespace StateNet.Tests.Network
 {
     public class StateNetworkBuilder_Tests
     {
+        private static ExpressionFactory<TransitionHistory> _expressions = new ExpressionFactory<TransitionHistory>();
+
+        private readonly TransitionHistory _context;
+
         [Fact]
-        public void Builder_Returns_FailedTransition_WhenStartStateIsNotSet()
+        public void Builder_Returns_FailMessage_WhenStartStateIsNotSet()
         {
             //Arrange
+            var networkBuilder = StateNetworkBuilder_Helpers.Empty_NetworkBuilder;
+            //Act
+            var sut = networkBuilder.Build();
 
+            //Assert
+            Assert.False(sut.Success);
+        }
+
+        [Fact]
+
+        public void StartState_IsSet_AfterBuild()
+        {
+            //Arrange
+            var networkBuilder = StateNetworkBuilder_Helpers.Empty_NetworkBuilder;
             //Act
-            var stateNetworkResult = NetworkBuilder.New.Build();
+            var sut = networkBuilder.SetStartState("A").Build();
 
             //Assert
-            Assert.False(stateNetworkResult.Success);
+            Assert.Equal("A", sut.Network.StartState);
         }
 
-        //public void Builder_Returns_FailedTransition_WhenNoConnectionsAreMadeSet()
+        [Fact]
+
+        public void AddState_SuccesfullyAddsState_AfterBuild()
+        {
+            //Arrange
+            var networkBuilder = StateNetworkBuilder_Helpers.SingleState_NetworkBuilder;
+            //Act
+            var sut = networkBuilder.AddState("b").Build();
+
+            //Assert
+            Assert.Contains("b", sut.Network.GetAllStates());
+        }
+
+        [Fact]
+
+        public void AddInput_SuccesfullyAddsInput_AfterBuild()
+        {
+            //Arrange
+            var networkBuilder = StateNetworkBuilder_Helpers.SingleState_NetworkBuilder; //Can now add input without a defined connection, so this doesn't have to be a valid network
+            //Act
+            var sut = networkBuilder.AddInput("1").Build();
+
+            //Assert
+            Assert.Equal("Unusable inputs exist in the network.", sut.Message);
```

Comment:
It would be worth while creating a 'Message Library' of sort. 
I.E a public static class which contains a bunch of properties like so:
public static readonly string UnsuableInput;

Otherwise these tests will all fail if we change the messages.